### ISSUE TITLE
Usage deprecated in pandas 0.23

### ISSUE THREAD

**Issue by pp-mo** [SciTools/iris#3018]
There was initially a failure due to pandas 0.22 -> 0.23 in #3017,
fixed by a tiny tweak.

However there is an outstanding deprecation warning emerging from the tests ..
```
$ python lib/iris/tests/test_pandas.py -v TestAsSeries.test_copy_float_false
test_copy_float_false (__main__.TestAsSeries) ... /home/h05/itpp/git/iris/iris_main/lib/iris/pandas.py:133: FutureWarning: Series.base is deprecated and will be removed in a future version
  if hasattr(pandas_obj, 'base'):
/home/h05/itpp/git/iris/iris_main/lib/iris/pandas.py:134: FutureWarning: Series.base is deprecated and will be removed in a future version
  base = pandas_obj.base
ok

----------------------------------------------------------------------
Ran 1 test in 0.004s

OK
```

That arises from the code [here](https://github.com/SciTools/iris/blob/ae81b7720a0ab6656d02291889eb6e190658acd9/lib/iris/pandas.py#L133)

This is not so trivial a problem : 
as existing code indicates, the usage of the 'base' property already changed.
If I understand right, potentially we can now just replace 'base' with 'values'.
 \- except that would no longer work with older versions of pandas.
However, right now, even *testing* for the presence of a '.base' attribute is triggering the deprecation warning.

**Comment by hdyson** [comment]
Apologies for butting in.  Isn't the answer here to use (and check for) df.values.base instead of df.base, which would, I think, work with both older versions and 0.23?

As for deprecating this module, while I have used it for using iris with UGRID data, it's a bit of a niche usage so I'm not sure I could make a compelling case against deprecation.

**Comment by pelson** [comment]
@hdyson - I'm not close enough to the pandas API to answer definitively, but that suggestion sounds sensible. Would be happy to merge if you want to submit a PR?

**Comment by hdyson** [comment]
@pelson A couple of quick questions here.  Has there been a resolution to the conversation about deprecating the cube to/from dataframe/series functionality?  And if it is to be deprecated, will this functionality leave iris before the switch over to a pandas version which will have an error rather than a deprecation warning?

For the fix itself, I'd be happy to provide it; I'm just trying to confirm we're not in "re-arranging deckchairs on the Titanic" territory.

**Comment by pelson** [comment]
Lol. Love the metaphor! No decision has been made really. It seems like one
of those corners of Iris that doesn't get any attention, and is therefore
ripe for consideration to be deprecated. No matter what though, code won't
be deleted until v3.

**Comment by rcomer** [comment]
Looks like this was fixed by #3079.
